pkg/workloads/cortex/tf_api/api.py

```diff
@@ -493,23 +510,22 @@ def start(args):
         if api.get("request_handler") is not None:
             local_cache["request_handler"] = ctx.get_request_handler_impl(api["name"])
 
-        if not util.is_resource_ref(api["model"]):
-            if not os.path.isdir(args.model_dir):
-                ctx.storage.download_and_unzip_external(api["model"], args.model_dir)
+        if not os.path.isdir(args.model_dir):
+            if util.is_resource_ref(api["model"]):
+                model_name = util.get_resource_ref(api["model"])
+                model = ctx.models[model_name]
+                ctx.storage.download_and_unzip(model["key"], args.model_dir)
+            else:
+                download_dir_external(ctx, api["model"], args.model_dir)
 
             if args.only_download:
```

Comment:
This might as well be indented left one level, in case we pass in args.only_download but the file is already present